src/runtime_src/core/tools/xbtracer/src/app/launcher.cpp

```diff
@@ -282,28 +524,89 @@ int create_child_proc_as_suspended(launcher& app)
 int resume_child_proc_and_wait_for_completion(launcher& app)
 {
   DWORD retval = 0;
+  int retcode = 0;
   if (ResumeThread(app.m_pi.hThread) < 0)
   {
     log_e("Failed to resume thread");
-    return -1;
+    retcode = -1;
+    goto l_exit;
   }
 
   // Wait for child process to finish
   if (WaitForSingleObject(app.m_pi.hProcess, INFINITE) != WAIT_OBJECT_0)
   {
     log_e("Waiting for child process failed");
-    return -2;
+    retcode = -2;
+    goto l_exit;
```

Comment:
This is not right.   I understand we clean up but it is not cool, and definitely not exception safe.  I don't have `log_e` handy but it's probably not nothrow ?  Are you sure all exit paths of this function will actually reach `l_exit:` ?

Reply:
Although log_e doesn't throw and all path were reaching l_exit:, to remove goto refactored the code.